### PR TITLE
Remove unused code and unused arguments from translator package

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: translator/trace/protospan_translation.go [hunk 1]
```diff
@@ -185,56 +179,11 @@ func AttributeArrayToSlice(attrArray pdata.AnyValueArray) []interface{} {
...
-				return desc.attrType
-			}
-		}
-	} else {
-		for _, desc := range attrValDescriptions {
-			if desc.regex.MatchString(value) {
-				return desc.attrType
-			}
+func DetermineValueType(value string) pdata.AttributeValueType {
+	for _, desc := range attrValDescriptions {
```
Comment (bogdandrutu) [inline translator/trace/protospan_translation.go:184]: Used only with `false` omitSimpleTypes

File: translator/trace/protospan_translation_test.go [hunk 2]
```diff
@@ -189,3 +191,39 @@ func constructTestAttributeSubarray() pdata.AttributeValue {
...
+		dest.UpsertInt(key, iVal)
+	case pdata.AttributeValueDOUBLE:
+		fVal, _ := strconv.ParseFloat(val, 64)
+		dest.UpsertDouble(key, fVal)
+	case pdata.AttributeValueBOOL:
+		bVal, _ := strconv.ParseBool(val)
+		dest.UpsertBool(key, bVal)
+	case pdata.AttributeValueMAP:
+		var attrs map[string]interface{}
+		err := json.Unmarshal([]byte(val), &attrs)
```
Comment (jcchavezs) [inline translator/trace/protospan_translation_test.go:208]: Shall we log this error?

Comment (bogdandrutu) [reply translator/trace/protospan_translation_test.go]: No more, code completely removed.

**Comment by jcchavezs** [comment]
LGTM

**Review by jrcamp — APPROVED** [review]

**Pull request merged**